architecturalOverview.md

```diff
@@ -0,0 +1,7 @@
+# Project Overview
+The purpose of the project is to provide clean and minimalistic implementations of basic algorithms and data structures in Python 3. The project is licensed under the MIT license which allows the use of the code for educational and commercial purposes. The project uses Sphinx to generate documentation, github actions for continuous delivery and flake8 for unit testing. The project is licensed under the permissive MIT license.
+
+![Alt text](./overview.png "A brief project overview")
+
+## A more detailed project explanation
```

Comment:
Could you add link to specific location when you're mentioning them. For instance: [test folder](tests/)

Reply:
Thank you. I've added this stuff.